pumps/mongo_test.go

```diff
@@ -32,6 +33,103 @@ func TestMongoPump_capCollection_Enabled(t *testing.T) {
 	}
 }
 
+func TestMongoPumpOmitIndexCreation(t *testing.T) {
+
+	c := Conn{}
+	c.ConnectDb()
+	defer c.CleanDb()
+
+	pump := newPump()
+	conf := defaultConf()
+
+	mPump := pump.(*MongoPump)
+	mPump.dbConf = &conf
+	record := analytics.AnalyticsRecord{
+		OrgID: "test-org",
+		APIID: "test-api",
+	}
+	records := []interface{}{record, record}
+
+	tcs := []struct {
+		testName             string
+		shouldDropCollection bool
+		Indexes              int
+		OmitIndexCreation    bool
+		dbType               MongoType
+	}{
+		{
+			testName:             "omitting index creation - StandardMongo",
+			shouldDropCollection: true,
+			Indexes:              1, //1 index corresponding to _id
+			OmitIndexCreation:    true,
+			dbType:               StandardMongo,
+		},
+		{
+			testName:             "not omitting index creation but mongo collection already exists - StandardMongo",
+			shouldDropCollection: false,
+			Indexes:              1, //1 index corresponding to _id
+			OmitIndexCreation:    false,
+			dbType:               StandardMongo,
+		},
+		{
+			testName:             "not omitting index creation but mongo collection doesn't exists - StandardMongo",
+			shouldDropCollection: true,
+			Indexes:              4, //1 index corresponding to _id + 3 from tyk
+			OmitIndexCreation:    false,
+			dbType:               StandardMongo,
+		},
+		{
+			testName:             "omitting index creation - DocDB",
+			shouldDropCollection: true,
+			Indexes:              1, //1 index corresponding to _id
+			OmitIndexCreation:    true,
+			dbType:               AWSDocumentDB,
+		},
+		{
+			testName:             "not omitting index creation but mongo collection already exists - DocDB",
+			shouldDropCollection: false,
+			Indexes:              4, //1 index corresponding to _id + 3 from tyk
+			OmitIndexCreation:    false,
+			dbType:               AWSDocumentDB,
+		},
+		{
+			testName:             "not omitting index creation but mongo collection doesn't exists - DocDB",
+			shouldDropCollection: true,
+			Indexes:              4, //1 index corresponding to _id + 3 from tyk
+			OmitIndexCreation:    false,
+			dbType:               AWSDocumentDB,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.testName, func(t *testing.T) {
+			mPump.dbConf.OmitIndexCreation = tc.OmitIndexCreation
+			mPump.dbConf.MongoDBType = tc.dbType
+			mPump.log = log.WithField("prefix", mongoPrefix)
+			mPump.connect()
+			defer c.CleanIndexes()
+
+			if tc.shouldDropCollection {
+				c.CleanDb()
+			}
+
+			if err := mPump.ensureIndexes(); err != nil {
+				t.Error("there shouldn't be an error ensuring indexes", err)
+			}
+
+			mPump.WriteData(context.Background(), records)
+			indexes, errIndexes := c.GetIndexes()
+			if errIndexes != nil {
+				t.Error("error getting indexes:", errIndexes)
+			}
+
+			if len(indexes) != tc.Indexes {
+				t.Error("index differential:", indexes)
```

Comment:
what does it means `index diferential`? can we change it to something more explanatory?

Reply:
Changed the message @sredxny 👍